### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by RobertHebel**
Deployed to https://beghp.github.io/gh-pages-rc-8<br>- use jest instead of mocha/chai<br>- increase test coverage<br>- remove unused `getSlidesBounds` and `touchmove` functions<br>

### CODE REVIEW

File: test/unit/carousel.test.js [hunk 1]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+      const wrapper = setup({
+        arrows: true,
+      });
+
+      expect(wrapper.find('.BrainhubCarouselItem').first().hasClass('BrainhubCarouselItem--active')).toEqual(true);
+
+      wrapper.find('.BrainhubCarousel__arrowRight').simulate('click');
+
+      expect(wrapper.find('.BrainhubCarouselItem').at(1).hasClass('BrainhubCarouselItem--active')).toEqual(true);
+    });
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:265]:
```suggestion
    });

```

File: test/unit/carousel.test.js [hunk 2]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+    it('changes slide on active arrow click', () => {
+      const wrapper = setup({
+        arrows: true,
+      });
+
+      expect(wrapper.find('.BrainhubCarouselItem').first().hasClass('BrainhubCarouselItem--active')).toEqual(true);
+
+      wrapper.find('.BrainhubCarousel__arrowRight').simulate('click');
+
+      expect(wrapper.find('.BrainhubCarouselItem').at(1).hasClass('BrainhubCarouselItem--active')).toEqual(true);
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:264]: is `.toEqual(true)` needed?
it doesn't sound like a valid human phrase

File: test/unit/carousel.test.js [hunk 3]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
-
-    carousel.instance().setState({
-      carouselWidth: 1000,
-      dragOffset: -2500,
-    });
-    expect(carousel.instance().getNearestSlideIndex()).to.equal(3);
+      expect(wrapper.find('.BrainhubCarouselItem').at(1).hasClass('BrainhubCarouselItem--active')).toEqual(true);
+    });
   });
+  describe('arrows', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:246]:
```suggestion

  describe('arrows', () => {
```

File: test/unit/carousel.test.js [hunk 4]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+          <div className={'first'}/>
+          <div/>
+          <div/>
+        </Carousel>,
+      );
+
+      expect(wrapper.find('.BrainhubCarouselItem--active').children().hasClass('first')).toEqual(true);
+    });
+  });
+  describe('dots', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:225]:
```suggestion

  describe('dots', () => {
```

File: test/unit/carousel.test.js [hunk 5]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+          <div/>
+          <div/>
+          <div className={'third'}/>
+        </Carousel>,
+      );
+
+      expect(wrapper.find('.BrainhubCarouselItem--active').children().hasClass('third')).toEqual(true);
+    });
+
+    it('Displays the first slide if the value is lower than 0', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:211]:
```suggestion
    it('displays the first slide if the value is lower than 0', () => {
```

File: test/unit/carousel.test.js [hunk 6]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+        >
+          <div/>
+          <div/>
+          <div className={'third'}/>
+        </Carousel>,
+      );
+
+      expect(wrapper.find('.BrainhubCarouselItem--active').children().hasClass('third')).toEqual(true);
+    });
+    it('Displays the last slide if the value is bigger than the number of slides', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:197]:
```suggestion

    it('Displays the last slide if the value is bigger than the number of slides', () => {
```

File: test/unit/carousel.test.js [hunk 7]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
   });
+  describe('Carousel value', () => {
+    it('sets carousel initial slide index', () => {
+      const wrapper = setup({
+        value: 2,
+      });
+
+      expect(wrapper.find('.BrainhubCarouselItem').at(2).hasClass('BrainhubCarouselItem--active')).toEqual(true);
+    });
+    it('Negative slide values for infinite option', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:183]:
```suggestion

    it('Negative slide values for infinite option', () => {
```

File: test/unit/carousel.test.js [hunk 8]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+      });
 
-    carousel.instance().setState({
-      carouselWidth: 1000,
-      dragOffset: -1001,
+      wrapper.unmount();
+
+      expect(clearInterval).toHaveBeenCalledTimes(1);
+    });
+    it('changes slides automatically', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:153]:
```suggestion

    it('changes slides automatically', () => {
```

File: test/unit/carousel.test.js [hunk 9]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+          2400: {
+            value: 2,
+          },
+        },
+      });
+
+      expect(wrapper.find('.BrainhubCarouselItem').first().prop('style').width).toEqual(`${declaredWidth}px`);
+    });
   });
+  describe('autoplay', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:140]:
```suggestion

  describe('autoplay', () => {
```

File: test/unit/carousel.test.js [hunk 10]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+          },
+          2400: {
+            infinite: true,
+          },
+        },
+      });
+
+      expect(wrapper.find('.BrainhubCarouselItem')).toHaveLength(21);
+    });
+    it(`uses default value if it's not declared in breakpoint`, () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:120]:
```suggestion

    it(`uses default value if it's not declared in breakpoint`, () => {
```

File: test/unit/carousel.test.js [hunk 11]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+        rtl: true,
+      });
 
-    const autoPlayValue = carousel.instance().getProp('slidesPerPage');
-    expect(autoPlayValue).to.equal(2);
+      expect(wrapper.find('.BrainhubCarousel--isRTL')).toHaveLength(1);
+    });
   });
+  describe('breakpoints', () => {
+    it('Picks up correct props for set window size', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:104]:
```suggestion
    it('picks up correct props for set window size', () => {
```

File: test/unit/carousel.test.js [hunk 12]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
 
-  it('slidesPerPage is equal 2 if window width is greater than 500', () => {
-    window.resizeTo(600, 600);
+      const item = wrapper.find('.BrainhubCarouselItem').first();
 
-    const carousel = setup();
+      expect(item.props().style).toHaveProperty('width', '250px');
+    });
+  });
+  describe('RTL', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:94]:
```suggestion

  describe('RTL', () => {
```

File: test/unit/carousel.test.js [hunk 13]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+        <Carousel/>,
+      );
 
-    const autoPlayValue = carousel.instance().getProp('autoPlay');
-    expect(autoPlayValue).to.equal(3000);
+      expect(wrapper.find('.BrainhubCarouselItem')).toHaveLength(0);
+    });
   });
+  describe('itemWidth', () => {
+    it('Overwrites carousel item width', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:84]:
```suggestion
    it('overwrites carousel item width', () => {
```

File: test/unit/carousel.test.js [hunk 14]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+        <Carousel
+          slides={[
+            <div key={0}/>,
+          ]}
+        />,
+      );
+
+      expect(wrapper.find('.BrainhubCarouselItem')).toHaveLength(1);
+    });
+    it('Renders carousel items when passed as a prop', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:75]:
```suggestion

    it('renders carousel items when passed as a prop', () => {
```

File: test/unit/carousel.test.js [hunk 15]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+      expect(wrapper.find('.BrainhubCarouselItem')).toHaveLength(3);
+    });
+    it('Renders additional clones when in infinite mode', () => {
+      const wrapper = setup({
+        infinite: true,
+      });
+
+      expect(wrapper.find('.BrainhubCarouselItem')).toHaveLength(21);
+    });
+    it('Renders carousel items when passed as a prop', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:64]:
```suggestion

    it('renders carousel items when passed as a prop', () => {
```

File: test/unit/carousel.test.js [hunk 16]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
 describe('Carousel', () => {
-  it('returns value of a prop', () => {
-    const carousel = setup();
+  describe('render slides', () => {
+    it('renders carousel items', () => {
+      const wrapper = setup();
+
+      expect(wrapper.find('.BrainhubCarouselItem')).toHaveLength(3);
+    });
+    it('Renders additional clones when in infinite mode', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:57]:
```suggestion

    it('renders additional clones when in infinite mode', () => {
```

File: src/components/Carousel.js [hunk 17]
```diff
@@ -330,23 +317,6 @@ class Carousel extends Component {
     }));
   };
 
-  /**
-   * Function handling touch move if drag has started. Sets dragOffset in the state.
-   * @param {event} e event
-   */
-  onTouchMove = e => {
```
Comment (piotr-s-brainhub) [inline src/components/Carousel.js:337]: are these removed methods unneeded?

File: test/unit/carousel.test.js [hunk 18]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
-  it('slidesPerPage is equal 2 if window width is greater than 500', () => {
-    window.resizeTo(600, 600);
+      const item = wrapper.find('.BrainhubCarouselItem').first();
 
-    const carousel = setup();
+      expect(item.props().style).toHaveProperty('width', '250px');
+    });
+  });
+  describe('RTL', () => {
+    it('Adds BrainhubCarousel--isRTL when in RTL mode', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:95]:
```suggestion
    it('adds BrainhubCarousel--isRTL when in RTL mode', () => {
```

File: test/unit/carousel.test.js [hunk 19]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+      const wrapper = setup({
+        arrowLeft: <div className="left" />,
+        arrowLeftDisabled: <div className="left-disabled" />,
+        arrowRight: <div className="right" />,
+        arrowRightDisabled: <div className="right-disabled" />,
+        addArrowClickHandler: true,
+      });
+
+      expect(wrapper.find('.left-disabled')).toHaveLength(1);
+      wrapper.find('.right').simulate('click');
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:276]:
```suggestion

      wrapper.find('.right').simulate('click');

```

File: test/unit/carousel.test.js [hunk 20]
```diff
@@ -23,95 +18,267 @@ window.resizeTo = (width, height) => {
...
+      expect(wrapper.find('.left-disabled')).toHaveLength(0);
+      expect(wrapper.find('.left')).toHaveLength(1);
+      wrapper.find('.right').simulate('click');
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:279]:
```suggestion

      wrapper.find('.right').simulate('click');

```

File: test/unit/carousel.test.js (hunk 2 above)
Comment (RobertHebel) [reply test/unit/carousel.test.js:264]: Change to toBeTruthy

File: src/components/Carousel.js (hunk 17 above)
Comment (RobertHebel) [reply src/components/Carousel.js:337]: I tested the deployed version and it seems it works as expected. Also tests e2e are passing so I guess everything works all right

File: test/unit/carousel.test.js [hunk 21]
```diff
@@ -23,95 +18,283 @@ window.resizeTo = (width, height) => {
...
+          <div/>
+          <div/>
+          <div className={'third'}/>
+        </Carousel>,
+      );
+
+      expect(wrapper.find('.BrainhubCarouselItem--active').children().hasClass('third')).toBeTruthy();
+    });
+
+    it('Displays the last slide if the value is greater than the number of slides', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:206]:
```suggestion
    it('displays the last slide if the value is greater than the number of slides', () => {
```

File: test/unit/carousel.test.js [hunk 22]
```diff
@@ -23,95 +18,283 @@ window.resizeTo = (width, height) => {
...
+
+    it('renders carousel items when passed as a prop', () => {
+      const wrapper = mount(
+        <Carousel/>,
+      );
+
+      expect(wrapper.find('.BrainhubCarouselItem')).toHaveLength(0);
+    });
   });
+  describe('itemWidth', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:86]:
```suggestion

  describe('itemWidth', () => {
```

File: test/unit/carousel.test.js [hunk 23]
```diff
@@ -23,95 +18,285 @@ window.resizeTo = (width, height) => {
...
+      expect(wrapper.find('.BrainhubCarouselItem').at(0).hasClass('BrainhubCarouselItem--active')).toBeTruthy();
+
+      jest.advanceTimersByTime(11);
+
+      wrapper.update();
+
+      expect(wrapper.find('.BrainhubCarouselItem').at(1).hasClass('BrainhubCarouselItem--active')).toBeTruthy();
+    });
   });
+  describe('Carousel value', () => {
```
Comment (piotr-s-brainhub) [inline test/unit/carousel.test.js:183]:
```suggestion

  describe('Carousel value', () => {
```

**Pull request merged**